pulser-core/pulser/sequence/sequence.py

```diff
@@ -1678,6 +1705,7 @@ def _add(
             self._phase_shift(
                 pulse.post_phase_shift, *last.targets, basis=basis
             )
+        self._add_slm_pulse(pulse, channel)
```

Comment:
I find this very misleading, it makes it seem like you are adding an SLM pulse every time. I would move some checks outside of the function to make clear this is only taking effect in rare occasions, eg (in line with my suggestion above)
```suggestion
        if self._slm_mask_dmm and self._schedule[self._slm_mask_dmm]._waiting_for_first_pulse:
            self._add_slm_pulse(pulse, channel)
```

Reply:
I was not a big fan of my name `add_slm_pulse` neither